lib/auth.js

```diff
@@ -0,0 +1,138 @@
+'use strict'
+
+var caseless = require('caseless')
+  , uuid = require('node-uuid')
+  , helpers = require('./helpers')
+
+var md5 = helpers.md5
+  , toBase64 = helpers.toBase64
+
+
+function Auth () {
+  this.hasAuth = undefined
```

Comment:
do these have any default values? for example, wouldn't `sentAuth` be false on creation? 

Also, a comment explaining what you're doing in situations like this is always helpful, especially when you want future devs to also define new properties here.
something like `// define all public properties here`